Examples/src/main/java/com/mapbox/vision/examples/CustomDetectionActivityKt.kt

```diff
@@ -0,0 +1,159 @@
+package com.mapbox.vision.examples
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.RectF
+import android.os.Bundle
+import com.mapbox.vision.VisionManager
+import com.mapbox.vision.mobile.core.interfaces.VisionEventsListener
+import com.mapbox.vision.mobile.core.models.AuthorizationStatus
+import com.mapbox.vision.mobile.core.models.Camera
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.mobile.core.models.FrameSegmentation
+import com.mapbox.vision.mobile.core.models.classification.FrameSignClassifications
+import com.mapbox.vision.mobile.core.models.detection.DetectionClass
+import com.mapbox.vision.mobile.core.models.detection.FrameDetections
+import com.mapbox.vision.mobile.core.models.position.VehicleState
+import com.mapbox.vision.mobile.core.models.road.RoadDescription
+import com.mapbox.vision.mobile.core.models.world.WorldDescription
+import com.mapbox.vision.performance.ModelPerformance
+import com.mapbox.vision.performance.ModelPerformanceConfig
+import com.mapbox.vision.performance.ModelPerformanceMode
+import com.mapbox.vision.performance.ModelPerformanceRate
+import java.nio.ByteBuffer
+import kotlin.math.pow
+import kotlin.math.sqrt
+import kotlinx.android.synthetic.main.activity_custom_detection.*
+
+class CustomDetectionActivityKt : BaseActivity() {
+
+    private var visionManagerWasInit = false
+
+    // VisionEventsListener handles events from Vision SDK on background thread.
+    private val visionEventsListener = object : VisionEventsListener {
+
+        override fun onAuthorizationStatusUpdated(authorizationStatus: AuthorizationStatus) {}
+
+        override fun onFrameSegmentationUpdated(frameSegmentation: FrameSegmentation) {}
+
+        override fun onFrameDetectionsUpdated(frameDetections: FrameDetections) {
+            // prepare camera frame first thing
+            val originalImage = frameDetections.frame.image
+            val frame = Bitmap.createBitmap(
+                originalImage.size.imageWidth,
+                originalImage.size.imageHeight,
+                Bitmap.Config.ARGB_8888
+            )
+            // prepare direct ByteBuffer that will hold camera frame data
+            val buffer = ByteBuffer.allocateDirect(originalImage.sizeInBytes())
+            // we need to lock pixels  explicitly because we will draw in another (main) thread
+            originalImage.lockPixels()
+            // associate underlying native ByteBuffer with our buffer
+            originalImage.copyPixels(buffer)
```

Comment:
Seems pixels can be released right here, after we copied them to the buffer?

Reply:
And if you extract image -> bitmap convertion code then it'll become even more clear

Reply:
Yes, you're right.